plugins/techdocs-backend/src/techdocs/stages/prepare/helpers.ts

```diff
@@ -93,3 +93,44 @@ export const checkoutGitRepository = async (
 
   return repositoryTmpPath;
 };
+
+// Could be merged with checkoutGitRepository
+export const checkoutGithubRepository = async (
+  repoUrl: string,
+): Promise<string> => {
+  const parsedGitLocation = parseGitUrl(repoUrl);
+
+  // Should propably not be hardcoded names of env variables, but seems too hard to access config down here
+  const user = process.env.GITHUB_PRIVATE_TOKEN_USER || '';
+  const token = process.env.GITHUB_PRIVATE_TOKEN || '';
+
+  const repositoryTmpPath = path.join(
+    // fs.realpathSync fixes a problem with macOS returning a path that is a symlink
+    fs.realpathSync(os.tmpdir()),
+    'backstage-repo',
+    parsedGitLocation.source,
+    parsedGitLocation.owner,
+    parsedGitLocation.name,
+    parsedGitLocation.ref,
+  );
+
+  if (fs.existsSync(repositoryTmpPath)) {
+    const repository = await Repository.open(repositoryTmpPath);
+    await repository.mergeBranches(
+      parsedGitLocation.ref,
+      `origin/${parsedGitLocation.ref}`,
+    );
+    return repositoryTmpPath;
+  }
+
+  if (user !== '' && token !== '') {
```

Comment:
in general, don't be too specific - just do `if (user && token)`, which also captures unexpected null/undefined

Reply:
Thanks for the advise! I'm new to TypeScript 🙂 https://github.com/spotify/backstage/pull/2247/commits/8fe0099c2c6fe5b36f511de403d39461f77553f3